pages/nft-book-store/status/[classId].vue

```diff
@@ -1195,4 +1324,55 @@ async function copyPurchaseLink () {
   })
 }
 
+function downloadAllPurchaseLinks () {
+  downloadFile({
+    data: purchaseLinks.value,
+    fileName: `${classId.value}_purchase_links.csv`,
+    fileType: 'csv'
+  })
+}
+
+function printAllQRCodes () {
+  try {
+    sessionStorage.setItem(
+      'nft_book_press_batch_qrcode',
+      convertArrayOfObjectsToCSV(purchaseLinks.value.map(({ channel, ...link }) => ({ key: channel, ...link })))
+    )
+    window.open('/batch-qrcode?print=1', 'batch_qrcode', 'popup,menubar=no,location=no,status=no')
```

Comment:
Note that in some cases `sessionStorage` is not passed to `window.open`, like setting `noreferer` etc